entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityOperationService.kt

```diff
@@ -58,6 +57,67 @@ class EntityOperationService(
         return BatchOperationResult(ArrayList(success), ArrayList(errors))
     }
 
+    /**
+     * Update a batch of [entities].
+     * Only entities with relations linked to existing entities will be updated.
+     *
+     * @return a [BatchOperationResult] with list of updated ids and list of errors (either not totally updated or
+     * linked to invalid entity).
+     */
+    fun update(entities: List<ExpandedEntity>): BatchOperationResult {
+        return entities.parallelStream().map { entity ->
+            updateEntity(entity)
+        }.collect(
+            { BatchOperationResult() },
+            { batchOperationResult, updateResult ->
+                updateResult.fold({
+                    batchOperationResult.errors.add(it)
+                }, {
+                    batchOperationResult.success.add(it)
+                })
+            },
+            BatchOperationResult::plusAssign
+        )
+    }
+
+    private fun updateEntity(entity: ExpandedEntity): Either<BatchEntityError, String> {
+        // All new attributes linked entities should be existing in the DB.
+        val linkedEntitiesIds = entity.getLinkedEntitiesIds()
+        val nonExistingLinkedEntitiesIds = linkedEntitiesIds
+            .minus(neo4jRepository.filterExistingEntitiesIds(linkedEntitiesIds))
```

Comment:
You could keep track of the existing entities to avoid querying them for each entity to update (you get them first in the handler)